src/gui/widgets/text_box_base.cpp

```diff
@@ -630,11 +660,12 @@ void text_box_base::signal_handler_sdl_key_down(const event::ui_event event,
 
 		case SDLK_RETURN:
 		case SDLK_KP_ENTER:
-			if(!is_composing() || (modifier & (KMOD_CTRL | KMOD_ALT | KMOD_GUI | KMOD_SHIFT))) {
-				return;
-			}
-			// The IME will handle it, we just need to make sure nothing else handles it too.
-			handled = true;
+
+//			if(!is_composing()) {
+//				return;
+//			}
```

Comment:
Some stuff to clean up

Reply:
Not really, somebody might need to check whether removing that check causes some sides effects as @CelticMinstrel has pointed out earlier, so I'll just add a TODO here. For now I can't find any such problems.